server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -42,54 +43,105 @@ class AddressController @Inject()(esRepo : ElasticsearchRepository)(implicit ec
   /**
     * Address query API
     *
-    * @param format
-    * @param input
+    * @param input  the address query
+    * @param format requested format of the query (paf/nag)
     * @return
     */
-  def addressQuery(
-    input : String,
-    format : String
-  ) : Action[AnyContent] = Action async {  implicit req =>
+  def addressQuery(input: String, format: String): Action[AnyContent] = Action async { implicit req =>
     logger info s"#addressQuery called with input $input , format: $format"
 
-    val regex : Regex = "(?:[A-Za-z]\\d ?\\d[A-Za-z]{2})|(?:[A-Za-z][A-Za-z\\d]\\d ?\\d[A-Za-z]{2})|(?:[A-Za-z]{2}\\d{2} ?\\d[A-Za-z]{2})|(?:[A-Za-z]\\d[A-Za-z] ?\\d[A-Za-z]{2})|(?:[A-Za-z]{2}\\d[A-Za-z] ?\\d[A-Za-z]{2})".r
-    val postcode : String = regex findFirstIn input getOrElse "Not recognised"
-    val buildingNumber : Int = input substring(0, 1) toInt
+    val regex: Regex = ("(?:[A-Za-z]\\d ?\\d[A-Za-z]{2})|(?:[A-Za-z][A-Za-z\\d]\\d ?\\d[A-Za-z]{2})|" +
+      "(?:[A-Za-z]{2}\\d{2} ?\\d[A-Za-z]{2})|(?:[A-Za-z]\\d[A-Za-z] ?\\d[A-Za-z]{2})|" +
+      "(?:[A-Za-z]{2}\\d[A-Za-z] ?\\d[A-Za-z]{2})").r
+    val tokens = AddressTokens(
+      uprn = "",
+      buildingNumber = input.substring(0, 2),
+      postcode = regex.findFirstIn(input).getOrElse("Not recognised")
+    )
+
+    logger info s"#addressQuery parsed: postcode: ${tokens.postcode} , buildingNumber: ${tokens.buildingNumber}"
 
-    logger info s"#addressQuery parsed: postcode: $postcode , buildingNumber: $buildingNumber"
+    format.stringToScheme() match {
+      case PostcodeAddressFile(str) => searchPafAddresses(tokens)
+      case BritishStandard7666(str) => searchUnsupportedFormatReply
+      case UnsupportedScheme(str) => searchUnsupportedFormatReply
+    }
+  }
 
-    Future fromTry format.toAddressScheme flatMap {
-      case PostcodeAddressFile(str) => esRepo.queryAddress(buildingNumber, postcode) map addressResponse
-      case _ => asyncWrongFormatResponse
+
+  private def searchPafAddresses(tokens: AddressTokens): Future[Result] = {
+    esRepo.queryAddress(tokens).map { addresses =>
+      Ok(Json.toJson(AddressBySearchResponseContainer(
+        AddressResponse(
+          tokens,
+          addresses = addresses.map(AddressResponseAddress.fromPafAddress),
+          limit = 10,
+          offset = 0,
+          total = addresses.size
+        ),
+        AddressResponseStatus.ok,
+        errors = Seq()
```

Comment:
just seeing empty seq's everywhere. for future reference default params are explicitly applied not implicitly applied